src/or/connection.c

```diff
@@ -113,11 +113,33 @@
 #include <sys/un.h>
 #endif
 
+/**
+ * On Windows and Linux we cannot reliably bind() a socket to an
+ * address and port if: 1) There's already a socket bound to wildcard
+ * address (0.0.0.0 or ::) with the same port; 2) We try to bind()
+ * to wildcard address and there's another socket bound to a
+ * specific address and the same port.
+ *
+ * To address this problem on these two platforms we implement a
+ * routine that:
+ * 1) Checks if first attempt to bind() a new socket  failed with
+ * EADDRINUSE.
+ * 2) If so, it will close the appropriate old listener connection and
+ * 3) Attempts bind()'ing the new listener socket again.
+ */
+#if defined(__linux__) || defined(_WIN32)
```

Comment:
If rebinding is a no-op when it's not required, let's do it on all platforms?

Reply:
Did so in 7c34404e3bd832d5ab5d5a2de6255dc05a3b790c.

Reply:
Fixed - 7c34404e3bd832d5ab5d5a2de6255dc05a3b790c